### PR TITLE
Material Inventory form

### PR DESCRIPTION
**Pull request by aidan-bowman**
## About the Pull Request

Automatically generated form for taking inventory of materials.

## Why It's Good For The Game

Muh bureaucracy

## Changelog

:cl:
add: The paperwork database now has a material inventory form (for logistics).
/:cl:

### CODE REVIEW

File: maps/site53/items/paperwork_dataserver.dm [hunk 1]
```diff
@@ -129,3 +129,36 @@ The entrant issued this permit is granted access to the zone listed.
...
+	. = ..()
+	stored_data = @{"[center][h1]Material Collection[/h1]
+[loglogo]
+[b]Secure. Contain. Protect.[/b][/center]
+[hr][table][row][cell][b]Material[/b][cell][b]Count[/b][cell][b]Type[/b][cell][b]Value[/b]
+"}
+
+	for(var/thing in SSmaterials.materials)
+		var/material/mat = thing
+		if(mat.hidden_from_codex || (mat.type == /material))
```
Comment (Tsurupeta) [inline maps/site53/items/paperwork_dataserver.dm]: can material be abstract????

**Comment by Tsurupeta** [comment]
also give me a screenshot of this form in game

**Comment by aidan-bowman** [comment]
Main problem with this (not the form's fault) is that there's a max cap of 20 fields, so the form doesn't fully work.

Easiest solution would just be to port better pencode that can handle more fields, although we *could* add a second pencode thing for a field that can be filled in and removed.

File: maps/site53/items/paperwork_dataserver.dm (hunk 1 above)
Comment (aidan-bowman) [reply maps/site53/items/paperwork_dataserver.dm]: Yep, done.

**Comment by aidan-bowman** [comment]
> also give me a screenshot of this form in game

```
[center][h1]Material Collection[/h1]
[loglogo]
[b]Secure. Contain. Protect.[/b][/center]
[hr][table][row][cell][b]Material[/b][cell][b]Count[/b][cell][b]Type[/b][cell][b]Value[/b][row][cell]placeholder[cell][field][cell][small]sheets[cell]1
[row][cell]steel[cell][field][cell][small]sheets[cell]4
[row][cell]plasteel[cell][field][cell][small]sheets[cell]12
[row][cell]titanium[cell][field][cell][small]sheets[cell]30
[row][cell]osmium-carbide plasteel[cell][field][cell][small]sheets[cell]12
[row][cell]sandstone[cell][field][cell][small]bricks[cell]4
[row][cell]marble[cell][field][cell][small]bricks[cell]20
[row][cell]concrete[cell][field][cell][small]bricks[cell]1
[row][cell]plastic[cell][field][cell][small]sheets[cell]1
[row][cell]wood[cell][field][cell][small]planks[cell]3
[row][cell]mahogany[cell][field][cell][small]planks[cell]45
[row][cell]maple[cell][field][cell][small]planks[cell]3
[row][cell]ebony[cell][field][cell][small]planks[cell]85
[row][cell]walnut[cell][field][cell][small]planks[cell]21
[row][cell]bamboo[cell][field][cell][small]planks[cell]3
[row][cell]yew[cell][field][cell][small]planks[cell]3
[row][cell]cardboard[cell][field][cell][small]sheets[cell]0
[row][cell]aluminium[cell][field][cell][small]sheets[cell]1
[row][cell]glass[cell][field][cell][small]sheets[cell]4
[row][cell]borosilicate glass[cell][field][cell][small]sheets[cell]40
[row][cell]uranium[cell][field][cell][small]sheets[cell]100
[row][cell]gold[cell][field][cell][small]ingots[cell]40
[row][cell]bronze[cell][field][cell][small]ingots[cell]40
[row][cell]copper[cell][field][cell][small]ingots[cell]1
[row][cell]silver[cell][field][cell][small]ingots[cell]35
[row][cell]osmium[cell][field][cell][small]ingots[cell]30
[row][cell]tritium[cell][field][cell][small]ingots[cell]300
[row][cell]deuterium[cell][field][cell][small]ingots[cell]1
[row][cell]metallic hydrogen[cell][field][cell][small]sheets[cell]100
[row][cell]platinum[cell][field][cell][small]ingots[cell]80
[row][cell]iron[cell][field][cell][small]ingots[cell]5
[row][cell]hematite[cell][field][cell][small]sheets[cell]1
[row][cell]rutile[cell][field][cell][small]sheets[cell]1
[row][cell]boron[cell][field][cell][small]sheets[cell]1
[row][cell]pitchblende[cell][field][cell][small]bricks[cell]1
[row][cell]graphite[cell][field][cell][small]sheets[cell]1
[row][cell]quartz[cell][field][cell][small]bricks[cell]1
[row][cell]pyrite[cell][field][cell][small]bricks[cell]1
[row][cell]spodumene[cell][field][cell][small]bricks[cell]1
[row][cell]cinnabar[cell][field][cell][small]bricks[cell]1
[row][cell]phosphorite[cell][field][cell][small]bricks[cell]1
[row][cell]rock salt[cell][field][cell][small]bricks[cell]1
[row][cell]potash[cell][field][cell][small]bricks[cell]1
[row][cell]bauxite[cell][field][cell][small]bricks[cell]1
[row][cell]sand[cell][field][cell][small]sheets[cell]1
[row][cell]phoron[cell][field][cell][small]crystals[cell]200
[row][cell]red[cell][field][cell][small]tiles[cell]1
[row][cell]diamond[cell][field][cell][small]gems[cell]70
[row][cell]nullglass[cell][field][cell][small]sheets[cell]1
[/table][hr][b]Inventory timestamp (date and time):[/b] [field]
[b]Signature of Officer Responsible for Inventory:[/b]
```

Capture1
Capture2

also FUCK is_abstract isn't working

**Comment by aidan-bowman** [comment]
ack materials can't be abstract because abstraction exists on datums and materials aren't datums

**Comment by Tsurupeta** [comment]
> ack materials can't be abstract because abstraction exists on datums and materials aren't datums

they are, anything /shit is a datum by default even if you dont specify datum keyword

**Comment by aidan-bowman** [comment]
> > ack materials can't be abstract because abstraction exists on datums and materials aren't datums
> 
> they are, anything /shit is a datum by default even if you dont specify datum keyword

mmm ok, fucking BYOND man I swear

anyways, the `placeholder` entry was apparently some *other* material subtype that wasn't removed from the codex for some reason

**Pull request merged**